docs/site/Where-filter.md

```diff
@@ -8,4 +8,560 @@ file: packages/metadata/README.md
 sidebar: lb4_sidebar
 permalink: /doc/en/lb4/Where-filter.html
 summary: A <i>where</i> filter specifies a set of logical conditions to match, similar to a WHERE clause in a SQL query.
----
\ No newline at end of file
+---
+
+## REST API
+
+In the first form below, the condition is equivalence, that is, it tests whether _property_ equals _value_.
+The second form below is for all other conditions.
+
+```
+filter[where][property]=value
+```
+
+```
+filter[where][property][op]=value
+```
+
+For example, if there is a cars model with an `odo` property, the following query finds instances where the `odo` is greater than 5000:
+
+```
+/cars?filter[where][odo][gt]=5000
+```
+
+For example, here is a query to find cars with `odo` is less than 30,000:
+
+```
+/cars?filter[where][odo][lt]=30000
+```
+
+Where:
+
+* _property_ is the name of a property (field) in the model being queried.
+* _value_ is a literal value. 
+* _op_ is one of the [operators](#operators) listed below.
+
+You can also use [stringified JSON format](Querying-data.html#using-stringified-json-in-rest-queries) in a REST query.
+
+### Filter limit
+
+{% include important.html content="There is a limit of twenty filters (combined with AND or OR) using this format, due to the use of [qs](https://github.com/ljharb/qs#parsing-arrays).  When there are more than twenty, the filter is converted into an `Object` where it is expecting
+an `Array`. See [LoopBack issue #2824](https://github.com/strongloop/loopback/issues/2824) for more details.
+" %}
+
+You can encode the large filter object as "stringified JSON":
+
+```
+http://localhost:3000/api/Books
+?filter={"where":{"or":[{"id":1},{"id":2},...,{"id":20"},{"id":21}]}}
+```
+
+<!-- **Override limit in `server.js`**
+
+```js
+// In `server/server.js`, before boot is called
+var loopback = require('loopback');
+var boot = require('loopback-boot');
+var qs = require('qs');
+
+var app = module.exports = loopback();
+app.set('query parser', function(value, option) {
+  return qs.parse(value, {arrayLimit: 500});
+});
+
+app.start = function() {
+  ...
+``` -->
+
+## Node API
+
+{% include content/angular-methods-caveat.html lang=page.lang %}
+
+### Where clause for queries
+
+For query methods such as `find()`,` findOrCreate()`, or `findOne()`, use the first form below to test equivalence, that is, whether _property_ equals _value_.
+Use the second form below for all other conditions.
+
+```ts
+{where: {property: value}} 
+```
+
+```ts
+{where: {property: {op: value}}}
+```
+
+Where:
+
+* _property_ is the name of a property (field) in the model being queried.
+* _value_ is a literal value. 
+* _op_ is one of the [operators](#operators) listed below.
+
+```ts
+await productRepository.find({where: {size:'large'}});
```

Comment:
Just a question.. in the controller, we can create `WhereBuilder` or `FilterBuilder` as an argument.  Does it work if we provide the JSON? If yes, perhaps we can list both? Again, I'm good to leave it out of scope for this PR and limit to "translating" the LB3 docs to LB4.

Reply:
I think it works with JSON. Will have examples in `where-filter.md` too.